src/event.rs

```diff
@@ -1055,12 +1055,21 @@ pub enum ElementState {
 }
 
 /// Describes a button of a mouse controller.
+///
+/// ## Platform-specific
+///
+/// **macOS:** does not offically define `Back` and `Forward`, but supports 29 generic mouse buttons from 3..=31.
+/// `3` and `4` is commonly used for `Back` and `Forward` respectivly by driver vendor and applications such as browsers,
+/// this follows this convention. Not guaranteed to work with all hardware and configurations.
```

Comment:
I think this is overly verbose for the users providing little value. I'd suggest it to add to macOS implementation instead in the `match` you've changed. And here I'd just say that it `might now work with all hardware`.